public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx

```diff
@@ -115,6 +115,7 @@ const OptionInput: FC<Props & { id: string; pathIndex?: string }> = ({
           {...register(name, {
             required: determineRequired(option, getValues, pathIndex),
             validate: (v) => (option.validationRule !== '' ? validateOption(v, option.validationRule) : true),
+            setValueAs: option.setValueAs,
```

Comment:
Have you tried something like this, to avoid the `any`? You might need to modify it a bit
```suggestion
            setValueAs: (val) =>  option.setValueAs(val),
```

Reply:
It would definitely satisfy Betterer but I'm not sure if trading explicit `any` for implicit `any` is actually valuable – I prefer your other suggestion to simply type stricter at the expense of additional maintenance if we ever add any additional input elements :D